File: remove_dupplicate.cpp
#include "leetcode.h"

class Solution {
public:
  int removeDuplicates(vector<int> &nums) {
    if (nums.size() == 0)
      return 0;
    int ans = 0;
    for (int i = 1; i < nums.size(); i++) {
      if (nums[ans] != nums[i])
        ans++;
      nums[ans] = nums[i];
    }
    return ans + 1;
  }
};

int main() {
  Solution obj;
  vector<int> nums = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
  cout << obj.removeDuplicates(nums);
}

File: other/remove_dupplicate.rs
struct Solution;

impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        match nums.is_empty() {
            true => 0,
            false => {
                let mut prev = 0;
                for i in 1..nums.len() {
                    if nums[prev] != nums[i] {
                        prev += 1;
                        nums[prev] = nums[i];
                    }
                }
                (prev + 1) as i32
            }
        }
    }
}

fn main() {
    let mut nums = vec![0,0,1,1,1,2,2,3,3,4];
    print!("{}", Solution::remove_duplicates(nums));
}
